source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -396,15 +396,29 @@ class ComponentInterface : public NodeT {
    * @brief Look up a transform from TF.
    * @param frame The desired frame of the transform
    * @param reference_frame The desired reference frame of the transform
-   * @param time_point The time at which the value of the transform is desired (default: 0, will get the latest)
+   * @param time_point The time at which the value of the transform is desired
    * @param duration How long to block the lookup call before failing
    * @throws modulo_components::exceptions::LookupTransformException if TF buffer/listener are unconfigured or
    * if the lookupTransform call failed
    * @return If it exists, the requested transform
    */
+   [[nodiscard]] state_representation::CartesianPose lookup_transform(
+      const std::string& frame, const std::string& reference_frame, const tf2::TimePoint& time_point,
+      const tf2::Duration& duration
+  );
+
+  /**
+   * @brief Look up a transform from TF.
+   * @param frame The desired frame of the transform
+   * @param reference_frame The desired reference frame of the transform
+   * @param validity_period The validity period of the latest transform from the time of lookup in seconds
+   * @param duration How long to block the lookup call before failing
+   * @throws modulo_components::exceptions::LookupTransformException if TF buffer/listener are unconfigured,
+   * if the lookupTransform call failed, or if the transform is too old
+   * @return If it exists and is still valid, the requested transform
+   */
   [[nodiscard]] state_representation::CartesianPose lookup_transform(
-      const std::string& frame, const std::string& reference_frame = "world",
-      const tf2::TimePoint& time_point = tf2::TimePoint(std::chrono::microseconds(0)),
+      const std::string& frame, const std::string& reference_frame = "world", double validity_period = -1.0,
```

Comment:
I am coming late to the party. I don't really like the mix between double and duration to represent the same thing, i.e. duration. I like to have proper unit than double values and you never really remember if it is seconds or milliseconds or event something else. With chrono literals it is still easy to fill those with `1s` or `200ms`.

Reply:
The only thing I see is that the `validity_period` could be `std::chrono` to enable the use of those literals and then `duration` a weird `tf2::Duration`. I would be tempted to put them both `std::chrono`. I just kept the duration initially as its ros type because it was the exact same signature than ros and if people are used to it then it make sense. But if we start changing it anyway then `std::chrono` are definitely more usable.

Reply:
Well you have a point there, mixing double and tf types in the same function is not very nice. Replacing the tf types would be nice but I wouldn't know how to do that for the `TimePoint`. The `Duration` is easier. What do you think @eeberhard ?